chat_server/sio/handlers/user_message.py

```diff
@@ -25,166 +25,140 @@
 # LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
 # NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
 # SOFTWARE,  EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
-from time import time
 
-from klatchat_utils.common import generate_uuid
 from klatchat_utils.database_utils.mongo_utils.queries import mongo_queries
-from klatchat_utils.database_utils.mongo_utils.queries.wrapper import MongoDocumentsAPI
+from klatchat_utils.database_utils.mongo_utils.queries.wrapper import (
+    MongoDocumentsAPI,
+)
 from klatchat_utils.database_utils.mongo_utils.structures import MongoFilter
 from neon_utils.logger import LOG
 from chat_server.sio.server import sio
 from chat_server.sio.utils import emit_error, login_required
 from chat_server.server_config import server_config
 from chat_server.utils.enums import UserRoles
 from chat_server.utils.services.popularity_counter import PopularityCounter
+from neon_data_models.models.api.klat.socketio import UserMessage
 
 
 @sio.event
 async def user_message(sid, data):
     """
     SIO event fired on new user message in chat
-    :param sid: client session id
+    :param sid: client session id (NOT shout_id which is in data['sid'])
     :param data: user message data
-    Example:
-    ```
-        data = {'cid':'conversation id',
-                'userID': 'emitted user id',
-                'promptID': 'id of related prompt (optional)',
-                'source': 'declared name of the source that shouted given user message'
-                'messageText': 'content of the user message',
-                'repliedMessage': 'id of replied message (optional)',
-                'bot': 'if the message is from bot (defaults to False)',
-                'lang': 'language of the message (defaults to "en")'
-                'attachments': 'list of filenames that were send with message',
-                'context': 'message context (optional)',
-                'test': 'is test message (defaults to False)',
-                'isAudio': '1 if current message is audio message 0 otherwise',
-                'messageTTS': received tts mapping of type: {language: {gender: (audio data base64 encoded)}},
-                'isAnnouncement': if received message is the announcement,
-                'timeCreated': 'timestamp on which message was created'}
-    ```
     """
-    LOG.info(f"Received user message data: {data}")
+    LOG.debug(f"Received user message data: {data}")
     try:
-        data["is_bot"] = data.pop("bot", "0")
-        data["context"] = data.get("context") or {}
-        is_bot = data["is_bot"] == "1"
-        is_proctor = False
-        if data["userID"].startswith("neon") and not is_bot:
+        data.setdefault("sid", "")  # TODO: This is patching clients that exclude `sid`
```

Comment:
`sid`, `message_id`, `messageID` are equivalent; are there still cases where none of those are set?

Reply:
https://github.com/NeonGeckoCom/neon-data-models/pull/29